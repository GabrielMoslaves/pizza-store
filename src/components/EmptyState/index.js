import React from "react";
import styles from "./styles..module.scss";

const EmptyState = () => {
  return (
    <div className={styles.container}>
      <h1>Sem produtos no carrinho</h1>
      <img width={200} src="/empty-cart.png" alt="carrinho-vazio" />
      <p>Adicione produtos e eles aparecer√£o aqui :)</p>
    </div>
  );
};

export default EmptyState;
