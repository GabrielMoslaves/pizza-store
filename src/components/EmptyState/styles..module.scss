.container {
  max-width: 319px;
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  flex-direction: column;
  background-color: #ffffff;
}
