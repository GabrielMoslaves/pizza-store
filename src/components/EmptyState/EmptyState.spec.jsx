import { render, screen } from "@testing-library/react";
import EmptyState from ".";

describe("<EmptyState/>", () => {
  it("should render emptystate correctly", () => {
    render(<EmptyState />);

    expect(screen.getByRole("heading", { name: /sem produtos no carrinho/i })).toBeInTheDocument();
    expect(screen.getByRole("img", { name: /carrinho-vazio/i })).toHaveAttribute(
      "src",
      "/empty-cart.png"
    );
    expect(screen.getByText(/Adicione produtos e eles aparecerÃ£o aqui/i)).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(<EmptyState />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
