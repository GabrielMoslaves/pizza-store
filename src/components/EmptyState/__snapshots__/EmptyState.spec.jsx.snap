// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyState/> should match snapshot 1`] = `
<div
  class="container"
>
  <h2>
    Sem produtos no carrinho
  </h2>
  <img
    alt="carrinho-vazio"
    src="/empty-cart.png"
    width="200"
  />
  <p>
    Adicione produtos e eles aparecer√£o aqui :)
  </p>
</div>
`;
